R/parseOutput.R

```diff
@@ -1845,58 +1845,49 @@ extractTech10 <- function(outfiletext, filename) {
     
-    if (grepl("^\\s{5}\\S", bivarFit[l], perl = TRUE)) {
-      # Parse new vars line
-      vars <- unlist(strsplit(trimSpace(bivarFit[l]), "\\s+", perl = TRUE))
-    }
-    else if (grepl("Bivariate (Pearson|Log-Likelihood) Chi-Square", bivarFit[l], perl = TRUE)) {
-      if (grepl("Overall", bivarFit[l], perl = TRUE)) { next } # Skip 'overall' values
-      
-      m <- unlist(regmatches(bivarFit[l], regexec("Bivariate (Pearson|Log-Likelihood) Chi-Square\\s+(\\S+)", bivarFit[l], perl = TRUE)))
-      
-      if (m[2] == 'Pearson') {
-        lastPearson <- m[3]
+    lines.idxs <- (v.line+1):end
+    
+    ldply(lines.idxs, function(line) {
+      if (grepl("^\\s+Category \\d+\\s+Category \\d+", bivarFit[line], perl = TRUE)) {
+        vals <- unlist(strsplit(trimSpace(bivarFit[line]), "\\s{2,}", perl = TRUE))
+        
+        c(vars, vals)
       }
-      else {
-        bivarFitStats <- rbind(bivarFitStats, c(vars, lastPearson, m[3]))
+      else if (grepl("^\\s+Bivariate (Pearson|Log-Likelihood) Chi-Square", bivarFit[line], perl = TRUE)) {
+        m <- unlist(regmatches(bivarFit[line], regexec("Bivariate (Pearson|Log-Likelihood) Chi-Square\\s+(\\S+)", bivarFit[line], perl = TRUE)))
+        
+        c(vars, "Summary", m[2:3], NA, NA)
       }
-    }
-    else {
-      values <- unlist(strsplit(trimSpace(bivarFit[l]), "\\s{2,}", perl = TRUE))
-      
-      bivarFitData[mPos,] <-c(vars,values)
-      mPos <- mPos + 1
-    }
-  }
+      else if (grepl("^\\s+Number of Significant Standardized Residuals", bivarFit[line], perl = TRUE)) {
+        m <- unlist(regmatches(bivarFit[line], regexec("Number of Significant Standardized Residuals\\s+(\\S+)", bivarFit[line], perl = TRUE)))
+        
+        c(vars, "Summary", "Significant", m[2], NA, NA)
+      }      
+    })
+  })
   
-  # Remove empty rows, and convert to data.frame
-  bivarFitData <- bivarFitData[rowSums(is.na(bivarFitData)) != ncol(bivarFitData),]
-  bivarFitData <- as.data.frame( bivarFitData, stringsAsFactors = FALSE )
-  names(bivarFitData) <- c("var1", "var2", "cat1", "cat2", "h0", "h1", "z")
+  bivarFitData <- res[res$V3 != "Summary",]
+  bivarFitStats <- res[res$V3 == "Summary", c("V1", "V2", "V4", "V5")]
   
-  # Fix data types
-  bivarFitData[,c("h0", "h1", "z")] <- as.numeric(unlist(bivarFitData[,c("h0", "h1", "z")]))
+  names(bivarFitData) <- c("var1", "var2", "cat1", "cat2", "h1", "h0", "z")
+  rownames(bivarFitData) <- NULL
   
-  bivarFitStats <- setNames(data.frame(bivarFitStats, stringsAsFactors = FALSE), c("var1","var2","pearson","log-likelihood"))
-  bivarFitStats[,c("pearson","log-likelihood")] <- as.numeric(unlist(bivarFitStats[,c("pearson","log-likelihood")]))
+  names(bivarFitStats) <- c("var1", "var2", "stat", "value")
+  bivarFitStats <- dcast(bivarFitStats, var1 + var2 ~ stat)
   
   tech10List$bivar_model_fit_info <- bivarFitData
-  tech10List$bivar_chi_square <- bivarFitStats
-  
+  tech10List$bivar_stats <- bivarFitStats
```

Comment:
Any concern about breaking backwards compatibility of MplusAutomation by changing the name of this output element from bivar_chi_square to bivar_stats?

Reply:
Done

Reply:
Yes, it's backwards incompatible, but it's an easy fix and the new name is a better description of what it contains. But happy to change it back if you'd prefer.

Reply:
From a reproducibility perspective, it's important that the name stays the same. There's also CRAN policy about anouncing breaking changes one major update ahead of time with a warning.. let's avoid all that and just keep the name the same.